ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarRecoveringCustodyImpl.java

```diff
@@ -235,59 +242,53 @@ private boolean isActiveSuperNode(final UInt64 slot) {
 
   private record RecoveryTask(
       AtomicReference<BeaconBlock> block,
-      Set<DataColumnSlotAndIdentifier> existingColumnIds,
+      Map<DataColumnSlotAndIdentifier, DataColumnSidecar> existingSidecars,
       AtomicBoolean recoveryStarted,
       AtomicBoolean timedOut) {}
 
   private void prepareAndInitiateRecovery(final RecoveryTask task) {
-    final SafeFuture<List<DataColumnSidecar>> list =
-        AsyncStream.createUnsafe(task.existingColumnIds().iterator())
-            .mapAsync(delegate::getCustodyDataColumnSidecar)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .toList();
-    initiateRecovery(task.block().get(), list);
+    final MetricsHistogram.Timer timer = dataAvailabilityReconstructionTimeSeconds.startTimer();
+    try {
+      initiateRecovery(task, task.existingSidecars.values(), timer);
+    } catch (final Exception e) {
+      task.recoveryStarted.compareAndSet(true, false);
```

Comment:
its far easier to reason about variables like this if they're set in a similar block, i'd be moving the set false to where it was set true